jobserver/management/commands/merge_projects.py

```diff
@@ -0,0 +1,79 @@
+import sys
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from jobserver.models import Project, User
+
+
+class MergeException(Exception):
+    pass
+
+
+@transaction.atomic()
+def merge_projects(username, primary_slug, other_slugs):
+    try:
+        operator = User.objects.get(username=username)
+    except User.DoesNotExist:
+        raise MergeException("User does not exist")
+
+    try:
+        primary = Project.objects.get(slug=primary_slug)
+    except Project.DoesNotExist:
+        raise MergeException("Primary project does not exist")
+
+    others = Project.objects.filter(slug__in=other_slugs)
+
+    if missing := set(other_slugs) - set(others.values_list("slug", flat=True)):
+        raise MergeException(f"Unknown projects: {', '.join(missing)}")
+
+    other_names = list(others.values_list("name", flat=True))
+
+    for project in others:
+        # set up redirect
+        primary.redirects.create(
+            created_by=operator,
+            old_url=project.get_absolute_url(),
+        )
+
+        # move workspaces
+        project.workspaces.update(project=primary)
```

Comment:
How will this handle workspaces with the same names?

Reply:
They're globally unique!  We use workspace names on the backends for the directories we run jobs inside of, and they predate projects as the "core" object.